File: leetcode/two_sum.go
// https://leetcode.com/problems/two-sum/


//func twoSum(nums []int, target int) []int {
//	m := make(map[int]int, len(nums))
//	for i, v := range nums {
//		m[v] = i
//	}
//	for v, i := range m {
//		other := target - v
//		if p, ok := m[other]; ok && other != v {
//			return []int{i, p}
//		}
//		if other == v {
//			for other_i := range nums {
//				if nums[other_i] == other && other_i != m[v] {
//					return []int{other_i, m[v]}
//				}
//			}
//		}
//	}
//	return []int{}
//}


func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, v := range nums {
		n, ok := m[v]
		m[target-v] = i
		if ok {
			return []int{i, n}
		}
	}
	return []int{}
}

File: python/two_sum.py
"""
https://leetcode.com/problems/two-sum/
"""


class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashmap = dict()
        for index, value in enumerate(nums):
            hashmap[value] = index
        for item in hashmap:
            other = target - item
            if other in hashmap and other != item:
                return [hashmap[item], hashmap[other]]
            if other == item:
                other_index = nums.index(other)
                if other_index != hashmap[item]:
                    return [other_index, hashmap[item]]
